packages/modal/src/ModalPanel.tsx

```diff
@@ -49,26 +51,22 @@ export const ModalPanel = ({ children, title, onDismiss }: ModalPanelProps) => {
 				}}
 				style={animationStyles}
 			>
-				<Flex justifyContent="flex-end">
-					<Button
-						variant="tertiary"
-						onClick={onDismiss}
-						iconAfter={CloseIcon}
-						css={{
-							position: 'relative',
-							top: `-${mapSpacing(1)}`,
-							right: `-${mapSpacing(1)}`,
-						}}
-					>
-						Close
-					</Button>
-				</Flex>
+				{title && <ModalTitle id={titleId}>{title}</ModalTitle>}
```

Comment:
Title should always exist, even if it's visually hidden otherwise there will be no `titleId` to point to